### PR TITLE
Add structs and Ballerina by Example for mutual ssl

### PR DESCRIPTION
**Pull request by Bhashinee**
Add structs to support mutual ssl in client side.
Add a ballerina by example for mutual ssl

### CODE REVIEW

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal [hunk 1]
```diff
@@ -11,6 +11,13 @@ public struct Response {
 public struct Options{
```
Comment (shafreenAnfar) [inline modules/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: Need a space here.

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java [hunk 2]
```diff
@@ -172,9 +176,43 @@ public HttpClientConnector getHTTPHttpClientConnector(String scheme, BConnector
         BStruct options = (BStruct) bConnector.getRefField(0);
         int followRedirect = options.getBooleanField(0);
         Long maxRedirectCount = options.getIntField(0);
+        String trustStoreFile = options.getStringField(0);
```
Comment (shafreenAnfar) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java]: Is it possible to have meaningful constants for these numbers ?

File: samples/ballerina-by-example/examples/mutual-ssl/server.bal [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    //Set trustStore password
+    trustStorePassword:"wso2carbon"
+}
+
+service<http> mutualSSL {
+    @http:resourceConfig {
+        methods:["GET"],
+        path:"/"
+    }
+
```
Comment (shafreenAnfar) [inline samples/ballerina-by-example/examples/mutual-ssl/server.bal]: Shall we remove this white space ?

File: samples/ballerina-by-example/examples/mutual-ssl/server.bal [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    resource sayHello (http:Request req, http:Response res) {
+        //Set response payload
+        response:setStringPayload(res, "Successful");
+        //Send response to client
+        response:send(res);
+    }
+
```
Comment (shafreenAnfar) [inline samples/ballerina-by-example/examples/mutual-ssl/server.bal]: This white space is also not needed..

File: samples/ballerina-by-example/examples/mutual-ssl/server.bal [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    keyStorePassword:"wso2carbon",
+    certPassword:"wso2carbon",
+    //To enable mutual ssl, have to configure the server by giving verifyClient = "require".
+    sslVerifyClient:"require",
+    //Set trustStore location.
+    trustStoreFile:"${ballerina.home}/bre/security/client-truststore.jks",
+    //Set trustStore password
+    trustStorePassword:"wso2carbon"
+}
+
```
Comment (shafreenAnfar) [inline samples/ballerina-by-example/examples/mutual-ssl/server.bal]: Let's remove this spacing ..

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java (hunk 2 above)
Comment (Bhashinee) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java]: Done

File: samples/ballerina-by-example/examples/mutual-ssl/server.bal (hunk 3 above)
Comment (Bhashinee) [reply samples/ballerina-by-example/examples/mutual-ssl/server.bal]: Done

File: samples/ballerina-by-example/examples/mutual-ssl/server.bal (hunk 4 above)
Comment (Bhashinee) [reply samples/ballerina-by-example/examples/mutual-ssl/server.bal]: Done

File: samples/ballerina-by-example/examples/mutual-ssl/server.bal (hunk 5 above)
Comment (Bhashinee) [reply samples/ballerina-by-example/examples/mutual-ssl/server.bal]: Done

File: samples/ballerina-by-example/examples/mutual-ssl/client.bal [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    http:Request req = {};
+    http:Response resp = clientConnector.get("/hello/", req);
+    system:println("Response code: " + response:getStatusCode(resp));
+    system:println("Response: " + response:getStringPayload(resp));
+}
+
+function getConnectorConfigs() (http:Options) {
+    http:Options option = {
+          ssl:{
+              //Set key store location
```
Comment (Bhashinee) [inline samples/ballerina-by-example/examples/mutual-ssl/client.bal]: Formatting issue fixed in https://github.com/ballerinalang/ballerina/pull/3627

File: samples/ballerina-by-example/examples/mutual-ssl/client.bal [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+              keyStoreFile:"${ballerina.home}/bre/security/wso2carbon.jks",
+              keyStorePassword:"wso2carbon",
+              //Set trust store location
+              trustStoreFile:"${ballerina.home}/bre/security/client-truststore.jks",
+              trustStorePassword:"wso2carbon"
+              }, followRedirects:{}
```
Comment (Bhashinee) [inline samples/ballerina-by-example/examples/mutual-ssl/client.bal]: Formatting issue fixed in #3627

**Pull request merged**